channeldb/db.go

```diff
@@ -137,7 +137,8 @@ type DB struct {
 	*bbolt.DB
 	dbPath string
 	graph  *ChannelGraph
-	now    func() time.Time
+
+	Now func() time.Time
```

Comment:
We already do it like this in two other locations. I didn't notice much problems with a closure really. As you say, if it is extended with additional functions like `time.After` it probably makes sense to have an interface

Reply:
One nice thing about using the interface that once it's spread, refactor will be simpler as well as it may lead to better code hygene (common test clock implementation maybe?)